packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -0,0 +1,195 @@
+    'bpk-sticky-chip-container',
+    `bpk-sticky-chip-container--${style}`,
+  );
+
+  const stickyChipClassnames = getClassName(
+    'bpk-sticky-chip',
+    stickyChip && stickyChip.className,
+  );
+
+  const scrollContainerLeftIndicator = getClassName(
+    'bpk-chip-group-scroller-left-indicator',
+    `bpk-chip-group-scroller-left-indicator--${style}`,
+  );
+
+  const scrollContainerRightIndicator = getClassName(
+    'bpk-chip-group-scroller-right-indicator',
+    `bpk-chip-group-scroller-right-indicator--${style}`,
+  );
+
+  const renderChipItem = ({ accessibilityLabel, component: Component = BpkSelectableChip, onClick, selected, text, ...rest }: ChipItem, index: number) => (
+    <Component
+      key={text}
+      selected={selected ?? false}
+      type={style}
+      accessibilityLabel={accessibilityLabel || text}
+      onClick={() => {
+        if (onClick) {
+          onClick(!selected, index);
+        }
+      }}
+      role="option"
+      {...rest}
+    >
+      {text}
+    </Component>
+  );
+
+  // TODO: Fix/remove the scroll indicator when style=CHIP_TYPES.onImage
+  // TODO: Fix shadows + focus indicators being cutoff when type = rail
+  const wrapRailInScroll = (children: ReactElement) =>
+    type === CHIP_GROUP_TYPES.rail ? (
+      <BpkMobileScrollContainer
+        scrollerRef={(el: HTMLElement) => {scrollContainerRef.current = el}}
+        leadingIndicatorClassName={isRTL() ? scrollContainerRightIndicator : scrollContainerLeftIndicator}
+        trailingIndicatorClassName={isRTL() ? scrollContainerLeftIndicator : scrollContainerRightIndicator}
+      >
+        {children}
+      </BpkMobileScrollContainer>
+    ) : children;
+
+  return (
+    <div className={containerClassnames}>
+      {type === CHIP_GROUP_TYPES.rail && (
+        <BpkBreakpoint query={BREAKPOINTS.ABOVE_TABLET}>
+          <Nudger leading chipStyle={style} scrollContainerRef={scrollContainerRef} />
+        </BpkBreakpoint>
+      )}
+      {type === CHIP_GROUP_TYPES.rail && stickyChip &&
+        <div className={stickyChipContainerClassnames}>
+          {renderChipItem({
+            ...stickyChip,
+            leadingAccessoryView: <FilterIconSm />,
+            className: stickyChipClassnames,
+          }, -1)}
+        </div>
+      }
+      {wrapRailInScroll(
+        <div className={chipGroupClassNames}
+         role="listbox"
+         aria-orientation="horizontal"
+         aria-multiselectable={!_isSingleSelect}
+         aria-label={ariaLabel}
+         aria-labelledby={ariaLabelledBy}
+        >
+          {chips.map((chip, index) => chip && renderChipItem(chip, index))}
+        </div>
+      )}
+      {type === CHIP_GROUP_TYPES.rail && (
+        <BpkBreakpoint query={BREAKPOINTS.ABOVE_TABLET}>
+          <Nudger chipStyle={style} scrollContainerRef={scrollContainerRef} />
+        </BpkBreakpoint>
+      )}
+    </div>
+  );
+};
+
+export const BpkChipGroupState = ({ chips, ...rest }: ChipGroupProps) => {
```

Comment:
The figma didn't mention that we'd expose stateful versions of these components but I needed them for testing anyway so they're here. If we'd prefer not to expose these then they can easily be moved to the example stories.

Reply:
I wonder if there is something more clever we can do here instead of exposing two different types of chips just a difference of having state.

What is the difference of needing to have a stateless and stateful chip? I think this should either be put to the consumer (like all our other components as Backpack shouldn't know about state) or built in mechanisms to detect to use state

Reply:
Decided to move the stateful ones to examples only as this aligns with the rest of backpack